### ISSUE TITLE
pragma STDC FP_CONTRACT ON in utilities.hpp is not supported in Visual Studio 2022

### ISSUE THREAD

**Issue by mdesmedt** [madmann91/bvh#43]
The default usage of `#pragma STDC FP_CONTRACT ON` in utilities.hpp causes warnings in MSVC 2022:
```
bvh\include\bvh/utilities.hpp(41,13): error C2220: the following warning is treated as an error
bvh\include\bvh/utilities.hpp(41,13): warning C4068: unknown pragma 'STDC'
bvh\include\bvh/utilities.hpp(50,13): warning C4068: unknown pragma 'STDC'
```

**Comment by madmann91** [comment]
The C++ standard states: ```Any pragma that is not recognized is ignored.```
This means that this code is fine. I don't have MSVC, nor do I plan on paying any Windows license to use that broken and poorly designed piece of software (there are other arguments to defend that position, namely that even on Windows, you have better options both in terms of standard compliance and performance of the generated code -- e.g. clang). I test that it works on Windows & MSVC using GitHub CI, but that's it. I don't have more time to invest with it, particularly if the problem is that you turned on the option to change warnings as errors. I already compile with `-Wextra -Wall -pedantic` on Linux, anything that pops up other than that is mostly junk coming from MSVC.

Want to fix it? Here are your options:
- Write a PR. Hint: MSVC has a non-standard pragma called `fp_contract` that does the same thing, or claims it does. I'll review your PR quickly and you can see if it works on other platforms with the integrated CI on GitHub, so you don't have to install a VM or any other operating system to test your fix.
- Turn off the option to emit warnings as errors in MSVC. Warnings will still be there. You can still fix them. They're just not errors.
- Use clang. For the reasons mentioned above, you'll get a significant performance boost out of that (plus support for OpenMP 3.0 which is from 2008, still not supported to this day in MSVC).

**Comment by madmann91** [comment]
Note that using `#pragma STDC FP_CONTRACT ON` the way the library does it is fine, as it is inside a compound statement (block) and appears before anything else (except conditional preprocessor directives). The documentation on Microsoft's webpage regarding the scope and meaning of `fp_contract` is lacking and it's unclear where the legal places for it are (but anyway, since it is non-standard, it can appear everywhere MSVC accepts it).

**Comment by tksuoran** [comment]
This message tells something about the scope for MSVC:

`#pragma 'fp_contract' should only be used at global scope or namespace scope`

This is the motivation to use namespace. I am updating the PR so that namespace is only used with MSVC.

**Comment by madmann91** [comment]
This should have been fixed by commit 5841efbf816a652ea0423862b14390ed8da3c4ad.
